### ISSUE TITLE
GCP PubSub Source fails after first create

### ISSUE THREAD

**Issue by dturanski** [knative/eventing-contrib#180]
## Steps to Reproduce the Problem

  1.
  2.
  3.

## Additional Info

**Comment by dturanski** [comment]
Actual Behavior:
Sample works as expected the first time. After deleting,  new GCP Pub Sub Source creation does not create a pod and source cannot deleted until I set `finalizers: []` in the yaml. 

Steps to Reproduce the Problem
1. Create a GCP Pub Sub Source and run the message dumper sample
2. Delete the source
3. Repeat

**Comment by vaikas-google** [comment]
Do you have any logs indicating what may have gone wrong?

**Comment by dturanski** [comment]
I'm not sure where to look but I was able to reproduce on 2 different clusters:
 `kubectl delete gcppubsubsources.sources.eventing.knative.dev/mysource` hangs until I remove the finalizers.  
The next time(s) creating the source never creates a pod. 
Do you have any suggestions of where to look?

**Comment by vaikas-google** [comment]
best place would be in the controller logs, which should land in:
kubectl -n knative-sources logs gcppubsub-controller-0

I'll try to repro myself or @Harwayne might get to it first.

**Comment by dturanski** [comment]
That's the thing. I don't see `gcppubsub-controller`.  Am I missing something? 
```
kubectl apply --filename https://github.com/knative/eventing-sources/releases/download/v0.2.1/release-with-gcppubsub.yaml
namespace/knative-sources unchanged
customresourcedefinition.apiextensions.k8s.io/containersources.sources.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/gcppubsubsources.sources.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/githubsources.sources.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/kuberneteseventsources.sources.eventing.knative.dev configured
clusterrole.rbac.authorization.k8s.io/manager-role configured
clusterrolebinding.rbac.authorization.k8s.io/manager-rolebinding configured
service/controller-manager-service unchanged
statefulset.apps/controller-manager unchanged
serviceentry.networking.istio.io/gcppubsub-bus-ext unchanged
```
```
$ kubectl get pod -n knative-sources
NAME                   READY   STATUS    RESTARTS   AGE
controller-manager-0   1/1     Running   2          3d
```

**Comment by dturanski** [comment]
I think the problem is that `knative/docs/eventing` is not up to date with `eventing-sources/contrib/gcppubsub`.

**Comment by vaikas-google** [comment]
Heh, yeah, probably :(

FYI:
https://github.com/knative/docs/pull/777

**Comment by vaikas-google** [comment]
Yes, so indeed, if it's the 0.2.1 it was part of the main one. If you wouldn't mind trying with the 3.0 and see if you run into similar issues. I just updated the readme (SORRY about the out of date docs :( )

**Comment by vaikas-google** [comment]
sigh upon re-reading, figured I'd clarify :) 0.2.1 gcp pubsub controller was part of the main controller for all the other sources. In 3.0 pubsub has its own controller and you should see two controllers in the knative-sources namespace.

**Comment by dturanski** [comment]
Sounds good.

**Comment by vaikas-google** [comment]
Just checking that the issue was actually resolved with the 0.3 release?

**Comment by n3wscott** [comment]
gcppubsubsource has been removed from this repo. Please see http://github.com/google/knative-gcp 

`PullSubscription` is the replacement. 

/close
